frontend/public/components/modals/roll-back-rc-modal.jsx

```diff
@@ -0,0 +1,107 @@
+import * as React from 'react';
+import { Trans, useTranslation } from 'react-i18next';
+import { createModalLauncher, ModalTitle, ModalBody, ModalSubmitFooter } from '../factory/modal';
+import { withHandlePromise } from '../utils';
+import { DeploymentConfigModel } from '../../models';
+import { k8sCreate, k8sUpdate } from '../../module/k8s';
+
+const BaseRollBackRCModal = withHandlePromise((props) => {
+  const { t } = useTranslation();
+  const dcName = props.resource.metadata.ownerReferences[0].name;
+  const [changeScaleSettings, setChangeScaleSettings] = React.useState(false);
+  const [changeStrategy, setChangeStrategy] = React.useState(false);
+  const [changeTriggers, setChangeTriggers] = React.useState(false);
+
+  const submit = (e) => {
+    e.preventDefault();
+
+    // put together a new rollback request
+    const req = {
+      kind: 'DeploymentConfigRollback',
+      apiVersion: 'apps.openshift.io/v1',
+      name: dcName,
+      latest: true,
+      force: true,
+      spec: {
+        from: {},
+        revision: parseInt(
+          props.resource?.metadata?.annotations?.['openshift.io/deployment-config.latest-version'],
+          10,
+        ),
+        includeTemplate: true,
+        includeReplicationMeta: changeScaleSettings,
+        includeStrategy: changeStrategy,
+        includeTriggers: changeTriggers,
+      },
+    };
+    const opts = {
+      name: dcName,
+      ns: props.resource.metadata.namespace,
+      path: 'rollback',
+    };
+
+    return props.handlePromise(
+      // create the deployment config rollback
+      k8sCreate(DeploymentConfigModel, req, opts).then((updatedDC) => {
+        // update the deployment config based on the one returned by the rollback
+        return k8sUpdate(DeploymentConfigModel, updatedDC);
+      }),
+      props.close,
+    );
+  };
+
+  return (
+    <form onSubmit={submit} name="form" className="modal-content">
+      <ModalTitle>{t('modal~Roll Back')}</ModalTitle>
+      <ModalBody>
+        <p>
+          <Trans t={t} ns="modal">
+            Use the following settings from{' '}
+            <strong className="co-break-word">
+              {{ resourceName: props.resource.metadata.name }}
+            </strong>{' '}
+            when rolling back:
+          </Trans>
+        </p>
+        <div className="checkbox">
+          <label className="control-label">
+            <input
+              type="checkbox"
+              onChange={() => setChangeScaleSettings(!changeScaleSettings)}
+              checked={!!changeScaleSettings}
+            />
+            {t('modal~replica count and selector')}
```

Comment:
Should these be sentence case?

Reply:
don't think so since there's a colon in initial message @bmignano could you verify?

Reply:
I consulted with Abi on the content side and she recommends sentence case for _everything_. So:

**Roll back**

Use the following settings from **dancer-ex-git-4** when rolling back:
[ ] Replica count and selector
[ ] Deployment strategy
[ ] Deployment trigger

Cancel | Roll back

Reply:
Great. @bmignano Can you confirm the screen shots above?

Reply:
In terms of capitalization, looks good! No need to combine **Roll back** into one word though (Modal header and confirmation button), unless that's more accurate from the technical side.

Reply:
I believe we decided above that it should be one word based on existing documentation.